esphome/components/qmc5883l/qmc5883l.cpp

```diff
@@ -60,10 +64,27 @@ void QMC5883LComponent::setup() {
     return;
   }
 
-  if (this->get_update_interval() < App.get_loop_interval()) {
+  uint32_t update_ms = this->get_update_ms_();
+  uint32_t loop_interval = App.get_loop_interval();
+  bool high_frequency = update_ms < loop_interval;
+  if (high_frequency)
     high_freq_.start();
+
+  ESP_LOGW(TAG, "interval = %u | loop_interval = %u | hf = %s", update_ms, loop_interval,
+           (high_frequency ? "true" : "false"));
+}
+
+void QMC5883LComponent::loop() {
+  // Changed ORIGINAL update() to read_data_().
+  //
+  // If DRDY Pin is Defined AND Signalled, Read Data; Otherwise, Handle in update()!
+  if (this->drdy_pin_ != nullptr) {
+    if (this->drdy_pin_->digital_read()) {
+      this->read_data_();
+    }
   }
```

Comment:
```suggestion
  this->read_data_();
```
Why not move this check into `read_data_` you can also call `this->disable_loop();` instead of using the pin as your switch between update and loop.

Reply:
I agree with all of your changes BESIDES this one...

I feel that adding an external function call into loop() to check DRDY in a secondary function adds unnecessary overhead.

Loop should be kept to a minimum!